homeassistant/components/edl21/sensor.py

```diff
@@ -0,0 +1,158 @@
+            if not obis:
+                continue
+
+            device = self._cache.get(obis)
+            if not device:
+                device = self._cache[obis] = EDL21Entity(obis, telegram)
+                new_devices.append(device)
+            elif device.update_telegram(telegram):
+                self._hass.async_create_task(device.async_update_ha_state())
+
+        if new_devices:
+            self._hass.async_add_job(
+                self._async_add_entities(new_devices, update_before_add=True)
+            )
+
+
+class EDL21Entity(Entity):
+    """Entity reading values from EDL21 telegram."""
+
+    # OBIS format: A-B:C.D.E*F
+    _OBIS_NAMES = {
+        # A=1: Electricity
+        # C=0: General purpose objects
+        "1-0:0.0.9*255": "Electricity ID",
+        # C=1: Active power +
+        # D=8: Time integral 1
+        # E=0: Total
+        "1-0:1.8.0*255": "Positive active energy total",
+        # E=1: Rate 1
+        "1-0:1.8.1*255": "Positive active energy in tariff T1",
+        # E=2: Rate 2
+        "1-0:1.8.2*255": "Positive active energy in tariff T2",
+        # D=17: Time integral 7
+        # E=0: Total
+        "1-0:1.17.0*255": "Last signed positive active energy total",
+        # C=15: Active power absolute
+        # D=7: Instantaneous value
+        # E=0: Total
+        "1-0:15.7.0*255": "Absolute active instantaneous power",
+        # C=16: Active power sum
+        # D=7: Instantaneous value
+        # E=0: Total
+        "1-0:16.7.0*255": "Sum active instantaneous power",
+        # A=129: Manufacturer specific
+        "129-129:199.130.3*255": "Manufacturer",
+        "129-129:199.130.5*255": "Public Key",
+    }
+
+    def __init__(self, obis, telegram):
+        """Initialize an EDL21Entity."""
+        self._obis = obis
+        self._telegram = telegram
+
+    @property
+    def should_poll(self) -> bool:
+        """Do not poll."""
+        return False
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._obis
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return a name."""
+        return self._OBIS_NAMES.get(self._obis)
+
+    @property
+    def state(self) -> str:
+        """Return the value of the last received telegram."""
+        value = self._telegram.get("value")
+        if value is None:
+            return STATE_UNKNOWN
+        return value
+
+    @property
+    def device_state_attributes(self):
+        """Enumerate supported attributes."""
+        attr = self._telegram.copy()
+        for key in ("objName", "unit", "value"):
+            if key in attr:
+                del attr[key]
+        return attr
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._telegram.get("unit")
+
+    @property
+    def icon(self):
+        """Return an icon."""
+        return ICON_POWER
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
This should not be throttled. It's being pushed to by events ?

Reply:
Yes. This was discussed above (in this PR). I can revert the commit, which introduced throttling, but the events arrive in intervals of a few seconds. Is it possible to throttle at a higher level to reduce the volume of the database?

Reply:
Right, this is my fault :wink:
I think if we don't hard code a throttle time, we at least have to make it configurable, as there are plenty of smart meters that report in 1s intervals and this generates a huge stress on the history DB (it made history unusable for me).

Reply:
Oh right. That's fair. I don't like limiting integrations because of the fragility of another integration 🤔

Reply:
So can we keep the code as is? Please mark this conversation as resolved in that case.